### DISCUSSION TITLE
Gatsby Image Incompatible Types

### DISCUSSION THREAD

**Discussion by bljdavidson** [gatsbyjs/gatsby#31088] [Help]
Hi,

The `getImage()` function returns the type `IGatsbyImageData | undefined`. However the image prop on the `<GatsbyImage />` component only excepts `IGatsbyImageData`. Because of this, I am getting the error:

```typescript
TS2322: Type 'IGatsbyImageData | undefined' is not assignable to type 'IGatsbyImageData'. Type 'undefined' is not assignable to type 'IGatsbyImageData'. 
```

Thanks,
Ben

**Comment by ascorbic** [comment]
Hi Ben. This is correct: you should always perform a null check on image data and display the image conditionally. e.g. 

```tsx

const image = getImage(data.image);

return <div>{ image ? <GatsbyImage image={image} /> : null }</div>

```
